src/net_processing.cpp

```diff
@@ -1700,7 +1702,11 @@ void PeerManagerImpl::RelayAddress(NodeId originator,
     // Use deterministic randomness to send to the same nodes for 24 hours
     // at a time so the m_addr_knowns of the chosen nodes prevent repeats
     uint64_t hashAddr = addr.GetHash();
-    const CSipHasher hasher = m_connman.GetDeterministicRandomizer(RANDOMIZER_ID_ADDRESS_RELAY).Write(hashAddr << 32).Write((GetTime() + hashAddr) / (24 * 60 * 60));
+
+    std::chrono::seconds current_time = GetTime<std::chrono::seconds>();
+    // Adding address hash makes exact rotation time different per address, while preserving periodicity.
+    uint64_t time_addr = static_cast<uint64_t>((current_time + std::chrono::seconds{static_cast<int64_t>(hashAddr)}) / ROTATE_ADDR_RELAY_DEST_INTERVAL);
```

Comment:
Can you explain why it is safe / defined behavior to perform a `static_cast<uint64_t>(std::chrono::seconds)`?  I can't figure out from reading https://en.cppreference.com/w/cpp/chrono/duration whether this is supposed to be defined, but maybe I'm missing something.

Reply:
Sorry for the noise, @sipa explained this to me (the result of the division is being cast, which is not a duration but a number).